### DISCUSSION TITLE
Configuration problem running dv8 showcase

### DISCUSSION THREAD

**Discussion by rnkazman** [sailuh/kaiaulu#334] [Q&A] [answered]
I received this error when running the dv8_showcase:

Error in parse_gitlog(perceval_path, git_repo_path) : 
  Unable to generate git log from this repository. Perhaps the path specified was incorrect or the repository has no commits?

I checked the value of perceval_path and it is correct.

The git_repo_path is:  "../../rawdata/pdfbox/git_repo/.git"

The showcase is being run in ~/Documents/GitHub/kaiaulu/kaiaulu
so the git_repo_path will turn into ~/Documents/GitHub/rawdata/pdfbox/git_repo/.git

But I was advised to create the rawdata folder within ~/Documents/GitHub/kaiaulu, which is what I did.  The folder is:
/Users/rnkazman/Documents/GitHub/kaiaulu/rawdata

Clearly these do not match.  What is the prefered solution?

**Comment by carlosparadis** [comment]
Not sure I follow the question. Are you asking why `../../` when it is `../`?

If so, this is a common point of confusion but it has less to do with Kaiaulu and more how it works when you are using the RStudio IDE. 

When you press play on the code block, you are executing the .Rmd file. That is on vignettes, hence ../../. However, if you copy and paste the code to the console, then you are running from whatever directory is specified, which generally is kaiaulu/kaiaulu. 

Or are you saying the config specifies to be in `~/Documents`?

**Reply by rnkazman** [reply]
I'm not sure I follow the answer.  :-)

I believe that I followed the folder conventions (that you helped me to organize), and I have not changed any path or code in dv8_showcase.Rmd, so why am I getting  this error?  The error indicates a path problem.  Hence my confusion.

I am attempting to execute these lines in the showcase:

```{r}
#git_repo_path <- "~/Downloads/testing-cochange/.git"
project_git <- parse_gitlog(perceval_path,git_repo_path)
project_git <- project_git  %>%
  filter_by_file_extension(file_extensions,"file_pathname")  %>% 
  filter_by_filepath_substring(substring_filepath,"file_pathname") %>% 
  filter_by_commit_size(commit_size = filter_commit_size)

kable(head(project_git))
```

and I get this error: 

Error in parse_gitlog(perceval_path, git_repo_path) : 
  Unable to generate git log from this repository. Perhaps the path specified was incorrect or the repository has no commits?

**Reply by carlosparadis** [reply, marked as answer]
Looking closely:

> The git_repo_path is: "../../rawdata/pdfbox/git_repo/.git"

Try to use instead: "../../rawdata/pdfbox/git_repo/pdfbox/.git"

Notice I added a `/pdfbox/` before the .git. When you git clone a project, you download both its source code, i.e. `/pdfbox/` and the git log, i.e. `/pdfbox/.git` inside of it. To double check the path, I suggest you go via terminal using cd inside the `.git` folder, and then type pwd. 

This is also what is used in Calculator: https://github.com/sailuh/kaiaulu/blob/ac522b6fd675230da905c0b54b241904e6c76705/conf/calculator.yml#L39 

I would in fact, recommend you try to run the notebook on Calculator first before trying to make your own on pdfbox just to get used to the path organization.

**Comment by rnkazman** [comment]
OK, it works now.  But, FYI, that was really confusing.  Why don't you create a standardized folder structure and then have a function that creates it?  Specifying paths like this is likely to be error-prone.  For example, I could specify the git repo (e.g. https://github.com/apache/pdfbox.git) and the function does the git clone and puts the data in the correct folder.  The learning curve is steep.  Why not make it a bit flatter?

**Reply by carlosparadis** [reply]
There is a function that creates the folder structure, but I did not have time to code review to merge it since the folder organization just got updated: https://github.com/sailuh/kaiaulu/pull/333 

I'm unsure if writing a git clone function in R would help here. There is no "correct folder". You must specify where the git log is, so Perceval can parse it. If you take Kaiaulu out of the equation, you must also tell Perceval where `.git` is. Same with DV8, too, neither download for you. 

If you want less control, then what you are looking for is the `/exec` scripts. Those use the low-level API you are using to facilitate some of the in-between steps. But this is also a work in progress. Some are pending review here: https://github.com/sailuh/kaiaulu/pull/329 

In essence, the architecture accommodates what you are looking for, but it has not been implemented yet.
